File: erpSoftExteriores/erpSoftExteriores.Web/Modules/Logs/LogRegistroSimple/LogRegistroSimple.cs

namespace erpSoftExteriores.Logs.Endpoints
{
    using Serenity;
    using Serenity.Data;
    using Serenity.Services;
    using System.Collections.Generic;
    using System.Data;
    using System.Web.Mvc;
    using MyRepository = Repositories.LogRegistroSimpleRepository;
    using MyRow = Entities.LogRegistroSimpleRow;


    public class LogRegistroSimple
    {        
        public static void CreateLogRegistroSimple(IUnitOfWork uow, long EntityId, string LocalTextPrefix)
        {
            try { 
            System.DateTime localDate = System.DateTime.Now;
            UserDefinition user = (UserDefinition)Authorization.UserDefinition;
            string entityType = LocalTextPrefix; 
            var Log = new MyRow();
            Log.FechaRegistro = localDate;
            Log.UsuarioRegistraId = user.UserId;
            Log.EntityType = entityType;
            Log.EntityId = EntityId;
            var LogsSaveRequest = new SaveRequest<MyRow> { Entity = Log };
            var response = new MyRepository().Create(uow, LogsSaveRequest);
            }
            catch { }
        }

        public static void UpdateLogRegistroSimple(IUnitOfWork uow,  object EntityId, string LocalTextPrefix)
        {
            try{ 
                System.DateTime localDate = System.DateTime.Now;
                UserDefinition user = (UserDefinition)Authorization.UserDefinition;
                string entityType = LocalTextPrefix;
                var fld = MyRow.Fields;
                var listRequest = new ListRequest
                {                
                    ColumnSelection = ColumnSelection.List,
                    EqualityFilter = new Dictionary<string, object>
                    {
                       { fld.EntityType.PropertyName, LocalTextPrefix },
                       { fld.EntityId.PropertyName, EntityId.ToString() }
                    }
                };
                var responseList = new MyRepository().List(uow.Connection, listRequest);
                var retrieveRequest = new RetrieveRequest();
                retrieveRequest.EntityId = responseList.Entities[0].Id;
                var responseRetrieve = new MyRepository().Retrieve(uow.Connection, retrieveRequest);
                responseRetrieve.Entity.FechaModifica = localDate;
                responseRetrieve.Entity.UsuarioModifica = user.UserId;
                //responseRetrieve.Entity.Texto1 = originalValues;
                var LogsUpdateRequest = new SaveRequest<MyRow> { Entity = responseRetrieve.Entity };
                var responseUpdate = new MyRepository().Update(uow, LogsUpdateRequest);
            }
            catch{ }
        }


        public static void DeleteLogRegistroSimple(IUnitOfWork uow, object EntityId, string LocalTextPrefix)
        {
            try
            {
                System.DateTime localDate = System.DateTime.Now;
                UserDefinition user = (UserDefinition)Authorization.UserDefinition;
                string entityType = LocalTextPrefix;
                var fld = MyRow.Fields;
                var listRequest = new ListRequest
                {
                    ColumnSelection = ColumnSelection.List,
                    EqualityFilter = new Dictionary<string, object>
                    {
                       { fld.EntityType.PropertyName, LocalTextPrefix },
                       { fld.EntityId.PropertyName, EntityId.ToString() }
                    }
                };
                var responseList = new MyRepository().List(uow.Connection, listRequest);
                var retrieveRequest = new RetrieveRequest();
                retrieveRequest.EntityId = responseList.Entities[0].Id;
                var responseRetrieve = new MyRepository().Retrieve(uow.Connection, retrieveRequest);
                //responseRetrieve.Entity.FechaModifica = localDate;
                //responseRetrieve.Entity.UsuarioModifica = user.UserId;
                responseRetrieve.Entity.Texto2 = "Deleted by " + user.Username + " on " + localDate;
                var LogsUpdateRequest = new SaveRequest<MyRow> { Entity = responseRetrieve.Entity };
                var responseUpdate = new MyRepository().Update(uow, LogsUpdateRequest);
            }
            catch { }
        }

        public RetrieveResponse<MyRow> Retrieve(IDbConnection connection, RetrieveRequest request)
        {
            return new MyRepository().Retrieve(connection, request);
        }

        public ListResponse<MyRow> List(IDbConnection connection, ListRequest request)
        {
            return new MyRepository().List(connection, request);
        }
    }
}

File: erpSoftExteriores/erpSoftExteriores.Web/Modules/Logs/LogRegistroSimple/LogRegistroSimple.ts
namespace erpSoftExteriores.Logs {

    export class LogRegistroSimple extends Serenity.Widget<any>{

        public container: JQuery;

        constructor(div: JQuery, item:Object) {
            super(div);
            this.setContainer(div);
            this.getLogData(item);
            
        }

        public setContainer(container) {
            this.container = container;
        }

        public getLogData(item) {

            var request: Serenity.ListRequest = {};
            request.Criteria = Serenity.Criteria.and(request.Criteria,
                [['EntityType'], '=', item.getEntityType()],
                [['EntityID'], '=', item.entityId]);

            Logs.LogRegistroSimpleService.SList(
                request,
                response => {
                    this.responseToTemplate(response)
                }
            );
        }


        public responseToTemplate(response: Serenity.ListResponse<any>) {
                if (response.TotalCount > 0) {

                var UsuarioModifica = response.Entities[0].UsuarioModificaUsername;
                var FechaModifica = response.Entities[0].FechaModifica;
                var UsuarioRegistra = response.Entities[0].UsuarioRegistraUsername;
                var FechaRegistro = response.Entities[0].FechaRegistro;

                FechaModifica = Q.formatDate(Q.parseISODateTime(FechaModifica), 'dd/MM/yyyy HH:mm tt');
                FechaRegistro = Q.formatDate(Q.parseISODateTime(FechaRegistro), 'dd/MM/yyyy HH:mm tt');

                var html = "<div><span>Creado el: <i>" + FechaRegistro + "</i></span><span>   por: <i><b>" + UsuarioRegistra +"</b></i></span>";
                if (FechaModifica) {
                    html = html + "  ||  <span>Modificado el: <i>" + FechaModifica + "</i></span><span>   por: <i><b>" + UsuarioModifica + "</b></i></span></div>";
                }
                else {
                    html = html + " </div>";
                }

                this.container.append(html);
            }




        }


    }
}